### PR TITLE
Tourniquets block medication until released

### LINKED ISSUE #2704: tourniquets have effect on injected medication
ACE3 Version: 3.3.2

Happens on Stable, Dev and RC Builds of ArmA 3

Mods:

@cba_a3
@ace3

Injected drugs still have full effect when applied to a limb with a tourniquet on it. I'm pretty sure if a tourniquet is applied correctly it would cut off all blood flow to that targeted limb, any drug applied to the targeted limb would only have local effects. I know the example I am about to list is a completely different drug, but the circulatory system is the highlight. 

May 3, 1995. Missouri. Emmitt Foster. Lethal Injection. Seven minutes after the lethal chemicals began to flow into Foster's arm, the execution was halted when the chemicals stopped circulating. . Death was pronounced thirty minutes after the execution began. According to William "Mal" Gum, the Washington County Coroner who pronounced death, the problem was caused by the tightness of the leather straps that bound Foster to the execution gurney; it was so tight that the flow of chemicals into the veins was restricted. Foster did not die until several minutes after a prison worker finally loosened the straps. The coroner entered the death chamber twenty minutes after the execution began, diagnosed the problem, and told the officials to loosen the strap so the execution could proceed.

source: http://www.deathpenaltyinfo.org/some-examples-post-furman-botched-executions

Steps to reproduce:
1) tourniquet a limb
2) apply morphine/epinephrine/atropin to the limb with a tourniquet on it

### PR DESCRIPTION
**Pull request by PabstMirror**
Close #2704

### CODE REVIEW

File: addons/medical/functions/fnc_actionRemoveTourniquet.sqf [hunk 1]
```diff
@@ -34,3 +35,20 @@ _target setVariable [QGVAR(tourniquets), _tourniquets, true];
 
 // Adding the tourniquet item to the caller
 _caller addItem "ACE_tourniquet";
+
+//Handle all injected medications now that blood is flowing:
+private _delayedMedications = _target getVariable [QGVAR(occludedMedications), []];
+private _removed = [];
+TRACE_2("meds",_part,_delayedMedications);
+{
```
Comment (jokoho48) [inline addons/medical/functions/fnc_actionRemoveTourniquet.sqf]: why you dont use the new select?

Comment (commy2) [inline addons/medical/functions/fnc_actionRemoveTourniquet.sqf]: Current Arma version is 1.54 and doesn't have select CODE

Comment (jokoho48) [inline addons/medical/functions/fnc_actionRemoveTourniquet.sqf]: but isnt 3.5.0 targeted for 1.56?

File: addons/medical/functions/fnc_actionRemoveTourniquet.sqf [hunk 2]
```diff
@@ -34,3 +35,20 @@ _target setVariable [QGVAR(tourniquets), _tourniquets, true];
...
+    _x params ["", "", "_medPartNum"];
+    if (_part == _medPartNum) then {
+        TRACE_1("delayed medication call after tourniquet removeal",_x);
+        ["medical_advMedication", [_target], _x] call EFUNC(common,targetEvent);
+        _removed pushBack _x;
```
Comment (Glowbal) [inline addons/medical/functions/fnc_actionRemoveTourniquet.sqf]: Why not replace the value by `objNull` and delete those after the for each? I'm not sure i Ilike the array for removal.

**Comment by Glowbal** [comment]
Nice work replacing those execRemote by events @PabstMirror :+1:

**Comment by Glowbal** [comment]
#3134

**Pull request merged**